src/main/java/com/microsoft/azure/relay/ClientWebSocket.java

```diff
@@ -36,6 +36,7 @@ public ClientWebSocket(TrackingContext trackingContext, AutoShutdownScheduledExe
 		this.fragmentQueue = new InputQueue<MessageFragment>(this.executor);
 		this.closeReason = null;
 		this.trackingContext = trackingContext;
+		this.socketLock = new AsyncLock(HybridConnectionListener.EXECUTOR);
```

Comment:
>HybridConnectionListener [](start = 34, length = 24)

I thought we ran into deadlock issues when the send side used the same thread pool as the receiver side. ClientWebSocket is used by both, correct?